common/crypto/attestation-api/test/get_attestation_app/app/main.cpp

```diff
@@ -35,28 +35,32 @@ int main()
         exit(-1);
     }
 
+    // set logging callback
+    logging_set_callback(puts);
```

Comment:
Should we maybe define `puts` as default in `common/logging/untrusted/logging.c`?   Note right now the callback is NULL by default and we do not test whether the callback is set, so we would crash  ...

Reply:
> Should we maybe define puts as default in common/logging/untrusted/logging.c?

No strong opinion on that.

With `puts`.
Pros: you should always get something displayed.
Cons: dependency on `puts`

Without `puts`.
Pros: no dependency on an output function.
Cons: if you forget to set it, nothing is displayed. 

> we do not test whether the callback is set, so we would crash

Nope, see line 26 of `common/logging/untrusted/logging.c`: if not set, the logging function returns 0/false.
Thing is that: the call is (usually) in a define, and the return value is not checked.

Reply:
ok, overlooked the check.  But is there any reason where `puts` is currently not available? Just seems to me having it as a default saves you having to add this in all test programs and if you forget you might be confused first at time you want to look at log and wouldn't see anything. I..e., i see only benefits but no real drawbacks

Reply:
> i see only benefits but no real drawbacks

In that case... done!